contracts/src/libraries/LPMath.sol

```diff
@@ -925,6 +950,48 @@ library LPMath {
             }
         }
 
+        // Simulate applying the share proceeds to the reserves and
+        // recalculate the present value.
+        bool success_;
+        (
+            _params.presentValueParams.shareReserves,
+            _params.presentValueParams.shareAdjustment,
+            _params.presentValueParams.bondReserves,
+            success_
+        ) = calculateUpdateLiquiditySafe(
+            _params.originalShareReserves,
+            _params.originalShareAdjustment,
+            _params.originalBondReserves,
+            _params.presentValueParams.minimumShareReserves,
+            -shareProceeds.toInt256()
+        );
+        if (!success_) {
+            // NOTE: If the updated reserves can't be calculated,  we can't
+            // continue the calculation. Return 0 to indicate that the share
+            // proceeds couldn't be calculated.
+            return 0;
+        }
+        uint256 presentValue_ = calculatePresentValue(
+            _params.presentValueParams
+        );
+
+        // Verify that the LP share price was conserved within a reasonable
+        // tolerance.
+        if (
+            // NOTE: Round up to make the check stricter.
+            presentValue_.divUp(_params.activeLpTotalSupply) +
```

Comment:
Thinking this is a similar issue to [here](https://github.com/spearbit-audits/delv-review/issues/30)....

here's how I think about it:

- increased strictness == more invalid values
- truthy check result == invalid value

Therefore `more truthy check results => more invalid values => increased strictness`

For this check, using `divDown` on the lhs and `divUp` on the rhs would result in the most truthy values